File: src/Neuralm.Presentation.Web/src/messages/responses/Response.ts
/**
 * Represents the Response class.
 */
export default class Response {
  public id: string;
  public requestId: string;
  public dateTime: Date;
  public message: string;
  public success: boolean;

  /**
   * Initializes a new instance of the Response class.
   */
  constructor(id: string, requestId: string, dateTime: Date, message: string, success: boolean) {
    this.id = id;
    this.requestId = requestId;
    this.dateTime = dateTime;
    this.message = message;
    this.success = success;
  }
}

File: src/Neuralm.Services/Neuralm.Services.Common.Messages/Abstractions/Response.cs
using Neuralm.Services.Common.Messages.Interfaces;
using System;

namespace Neuralm.Services.Common.Messages.Abstractions
{
    /// <summary>
    /// Represents the <see cref="Response"/> class.
    /// </summary>
    public abstract class Response : IResponseMessage
    {
        /// <summary>
        /// Gets the id.
        /// </summary>
        public Guid Id { get; set; }

        /// <summary>
        /// Gets the request id.
        /// </summary>
        public Guid RequestId { get; set; }

        /// <summary>
        /// Gets the date time.
        /// </summary>
        public DateTime DateTime { get; set; }

        /// <summary>
        /// Gets the message.
        /// </summary>
        public string Message { get; set; }

        /// <summary>
        /// Gets a value indicating whether the response was successful.
        /// </summary>
        public bool Success { get; set; }

        /// <summary>
        /// Initializes an instance of the <see cref="Response"/> class.
        /// </summary>
        /// <param name="requestId">The request id.</param>
        /// <param name="message">The message.</param>
        /// <param name="success">The success flag.</param>
        protected Response(Guid requestId, string message, bool success)
        {
            Id = Guid.NewGuid();
            RequestId = requestId;
            DateTime = DateTime.UtcNow;
            Message = message;
            Success = success;
        }

        /// <summary>
        /// Initializes a new instance of the <see cref="Response"/> class.
        /// SERIALIZATION CONSTRUCTOR.
        /// </summary>
        protected Response()
        {
            Id = Guid.NewGuid();
            DateTime = DateTime.UtcNow;
        }
    }
}
